### ISSUE TITLE
What type is ID in `T get(T : Object, ID)(ID id)`

### ISSUE THREAD

**Issue by vra5107** [buggins/hibernated#59]
What type is ID ? I am not able to find any thing called ID in HibernateD source code. It fails if I pass in an integer. 

```
T get(T : Object, ID)(ID id) 
T load(T : Object, ID)(ID id) 
void load(T : Object, ID)(T obj, ID id)

```

**Comment by bausshf** [comment]
AFAIK `ID` is a template type that can be any datatype that an entity allows. For ex. `get`, t's simply used in an internal query to retrieve the entity of type `T`.

```
T get(T : Object, ID)(ID id);
```

Calls

```
override Object getObject(string entityName, Variant id);
```

Which calls:

```
Object getObject(const EntityInfo info, Object obj, Variant id) {
        string hql = "FROM " ~ info.name ~ " WHERE " ~ info.getKeyProperty().propertyName ~ "=:Id";
        Query q = createQuery(hql).setParameter("Id", id);
        Object res = q.uniqueResult(obj);
        return res;
    }
```

Could you please provide an example of what exactly goes what and what you're trying to achieve? That way it will be easier to answer what you're supposed to provide or to tell whether there is a bug or not.

**Comment by vra5107** [comment]
Thankyou for your reply. I didn't see your message. I gave up on this and started using ddbc instead. Honestly I can't remember exactly what the use case was as I reverted most of HibernateD source code and swapped with ddbc. But my guess was I was trying to get an object with it's id and I passed in an integer.

I guess `ID` didn't strike me as a template type since it has two characters. Forgot this isn't Java (Apparantly even in Java it is just a convention).

I would love to switch back to HibernateD eventually. So many lines of code writting using ddbc just doesn't look good. 

I will close the issue now. Will reopen it if I see any issues.
